backends/dpdk/dpdkProgram.cpp

```diff
@@ -113,9 +113,10 @@ const IR::DpdkAsmProgram *ConvertToDpdkProgram::create(IR::P4Program *prog) {
     for (auto kv : structure->parsers) {
         if (kv.first == "IngressParser")
             kv.second->apply(*ingress_parser_converter);
-        else if (kv.first == "EgressParser")
-            kv.second->apply(*egress_parser_converter);
-        else if (kv.first == "MainParserT")
+        else if (kv.first == "EgressParser") {
```

Comment:
this is a bit strange.
First, because you compile for PSA, you should always have an egress.
Second, by giving a flag you are causing a large part of the P4 program to be ignored. I would at least check that the egress programs are "empty" and give a warning. Alternatively, you could dispense with the flag and not generate any code if all the egress components are empty. In other words, infer the value of this flag from the source program.

Reply:
Currently, code generated for ingress pipeline and egress pipeline are in single output file which is not right, because two should be independent.
Real future job is to generate two output file one for egress pipeline and other for egress pipeline from a single p4 program which has non empty ingress and egress block. Until this done we generate code for ingress only even when egress are present, but once above task done, we will get rid of this flag, that's why the flag is not exposed to user, it's hidden.

Reply:
it does not look very hidden to me.
Anyway, if this is temporary this should also be documented.